index.js

```diff
@@ -1,134 +1,225 @@
 'use strict'
 
 const fp = require('fastify-plugin')
-const { Cache } = require('async-cache-dedupe')
```

Comment:
note: I removed the module `async-cache-dedupe` because it's hard to delegate the ttl management to it
see below how it is replaced

Reply:
I'm not really comfortable with this choice :(. That module was tested specifically to avoid memory leaks and other problems related to the dedupe layer. By breaking its tight encapsulation we make it hard to test it and avoid leaks.

An alternative option is to ember that module here and iterate on it.

Reply:
update `async-cache-dedupe` and remove the ttl management could be a solution?

Reply:
Yes, definitely.

Reply:
Great! I'll go for that!

Reply:
Have you seen the options?